### PR TITLE
Dice coefficient update

### PR DESCRIPTION
**Pull request by bdnorman**
Should use floats for calculating dice coefficient since you are dividing intersection by union.

### CODE REVIEW

**Comment by jakeret** [comment]
I haven't had the time to check but I would have assumed that we have a float division because of the `from __future__ import division` at the top. Or am I missing something?

**Comment by bdnorman** [comment]
My mistake, I must have missed that at the top. That being said, dice coefficient is a calculation of intersection between two sets which you're not doing since you're feeding in the softmax scaled logits. The input predictions should be binary to compute the intersection with the truth value. I have tried implementing this by taking `tf.argmax(logits, axis = -1)` but tf.argmax does not have a gradient defined for backpropogation. Bottom line, I'm not sure how to correctly implement dice coefficient with tensorflow but I thought I'd bring it to your attention

**Comment by jakeret** [comment]
I thought about the `tf.argmax`... 
Anyway, I just merged a branch that includes a new implementation of the dice coefficient, which hopefully resolves the issues
